routes/api.js

```diff
@@ -431,7 +433,13 @@ router.get('/create_account', async (req, res) => {
         const user = await req.db.users.findOne({ where: { email: decoded.email } });
         if (!user) {
           res.status(400).json({ error: 'error_api_user_exists_not' });
+        } else if (user.creation_hash !== null) {
+          res.status(400).json({ error: 'error_api_account_creation_progress' });
         } else if (user.status === 'approved') {
+          const creationHash = hash.sha256(crypto.randomBytes(32)).toString('hex');
+          await req.db.users.update({
+            creation_hash: creationHash,
+          }, { where: { email: decoded.email } });
           // eslint-disable-next-line camelcase
           const { username, public_keys } = req.query;
           const weightThreshold = 1;
```

Comment:
before line 464 below do something like this to catch any race condition https://github.com/steemit/condenser/blob/master/src/server/api/general.js#L182-L195